File: python/baiduads-sdk/test/videoupload/video_upload_service_test.py
"""
for test VideoUploadService
"""
import time
import unittest

from baiduads.common.model import ApiRequestHeader
from baiduads_manual.videoupload.api.video_upload_service import VideoUploadService
from baiduads_manual.videoupload.model.add_video_params import AddVideoParams
from baiduads_manual.videoupload.model.add_video_request import AddVideoRequest
from baiduads_manual.videoupload.model.add_video_request_wrapper import AddVideoRequestWrapper


class TestVideoUploadService(unittest.TestCase):

    def setUp(self):
        self.api = VideoUploadService()

    def tearDown(self):
        pass

    def test_add_video(self):
        file = "/tmp/test2.mp4"
        # 请填写自己的信息
        header = ApiRequestHeader(
            token="xxxxx",
            username="xxxxx",
            password="xxxxx",
            target="xxxxx",
            _spec_property_naming=True
        )
        body = AddVideoRequest(
            file=file,
            signature="8e26c5df42a9efe908370591a670b358",
            params=AddVideoParams(_format="mp4", source=2, videoName="test" + str(time.time()) + ".mp4",
                                  httpProtocol="HTTP")
        )
        add_video_request_wrapper = AddVideoRequestWrapper(
            header=header,
            body=body
        )
        res = self.api.add_video(add_video_request_wrapper)
        assert res.header.status == 0
        print(res)


if __name__ == '__main__':
    unittest.main()

File: java/baiduads-sdk/src/test/com/baidu/dev2/api/sdk/videoupload/VideoUploadServiceTest.java
package com.baidu.dev2.api.sdk.videoupload;

import com.baidu.dev2.api.sdk.common.ApiRequestHeader;
import com.baidu.dev2.api.sdk.manual.upload.model.MultiUploadOptions;
import com.baidu.dev2.api.sdk.manual.videoupload.api.VideoUploadService;
import com.baidu.dev2.api.sdk.manual.videoupload.model.AddVideoParams;
import com.baidu.dev2.api.sdk.manual.videoupload.model.AddVideoRequest;
import com.baidu.dev2.api.sdk.manual.videoupload.model.AddVideoRequestWrapper;
import com.baidu.dev2.api.sdk.manual.videoupload.model.AddVideoResponseWrapper;
import org.apache.commons.codec.digest.DigestUtils;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import java.io.File;
import java.io.FileInputStream;

public class VideoUploadServiceTest {
    private VideoUploadService service;
    private ApiRequestHeader header;

    @Before
    public void before() {
        service = new VideoUploadService();
        header = new ApiRequestHeader();
        // 请填写自己的信息
        header.setAccessToken("xxxxx"); // 替换为授权生成的accessToken
        header.setUserName("xxxxx"); // 替换为自己的用户名
    }

    @Test
    public void testAddVideo() throws Exception {
        AddVideoRequestWrapper addVideoRequestWrapper = new AddVideoRequestWrapper();
        addVideoRequestWrapper.setHeader(header);

        File file = new File("/tmp/test2.mp4");
        String md5Str = DigestUtils.md5Hex(new FileInputStream(file));

        AddVideoRequest addVideoRequest = new AddVideoRequest();
        addVideoRequest.setFile(file);
        addVideoRequest.setSignature(md5Str);

        AddVideoParams addVideoParams = new AddVideoParams();
        addVideoParams.setVideoName("test" + System.currentTimeMillis() + ".mp4");
        addVideoParams.setHttpProtocol("HTTP");
        addVideoParams.setSource(2);
        addVideoParams.setFormat("mp4");
        addVideoRequest.setParams(addVideoParams);

        addVideoRequestWrapper.setBody(addVideoRequest);

        AddVideoResponseWrapper addVideoResponseWrapper = service.addVideo(addVideoRequestWrapper);
        System.out.println(addVideoResponseWrapper);
        Assert.assertTrue(addVideoResponseWrapper.getHeader().getStatus() == 0);
    }


    @Test
    public void testAddVideoMultiupload() throws Exception {
        AddVideoRequestWrapper addVideoRequestWrapper = new AddVideoRequestWrapper();
        addVideoRequestWrapper.setHeader(header);

        File file = new File("/tmp/test2.mp4");
        String md5Str = DigestUtils.md5Hex(new FileInputStream(file));

        AddVideoRequest addVideoRequest = new AddVideoRequest();
        addVideoRequest.setFile(file);
        addVideoRequest.setSignature(md5Str);

        AddVideoParams addVideoParams = new AddVideoParams();
        addVideoParams.setVideoName("test" + System.currentTimeMillis() + ".mp4");
        addVideoParams.setHttpProtocol("HTTP");
        addVideoParams.setSource(2);
        addVideoParams.setFormat("mp4");
        addVideoRequest.setParams(addVideoParams);

        MultiUploadOptions options = new MultiUploadOptions();
        options.setMultipartThreshold(100 * 1024); // 100kb
        addVideoRequest.setUploadOptions(options);


        addVideoRequestWrapper.setBody(addVideoRequest);

        AddVideoResponseWrapper addVideoResponseWrapper = service.addVideo(addVideoRequestWrapper);
        System.out.println(addVideoResponseWrapper);
        Assert.assertTrue(addVideoResponseWrapper.getHeader().getStatus() == 0);
    }
}
